webapp/javascript/components/TimelineChartWrapper.tsx

```diff
@@ -242,23 +228,121 @@ class TimelineChartWrapper extends React.Component<
         data-testid={this.props['data-testid']}
         id={id}
         options={customFlotOptions}
-        viewSide={viewSide}
-        data={flotData}
+        data={data}
+        //        data={d}
         width="100%"
         height="100px"
       />
     );
   };
 }
 
-function decodeTimelineData(timelineData: Timeline) {
-  if (!timelineData) {
+function isSingleDatapoint(timelineA: TimelineData, timelineB?: TimelineData) {
+  const aIsSingle = timelineA.data && timelineA.data.samples.length <= 1;
+  if (!aIsSingle) {
+    return false;
+  }
+
+  if (timelineB && timelineB.data) {
+    return timelineB.data.samples.length <= 1;
+  }
+
+  return true;
+}
+
+function skewTimeline(
+  timeline: TimelineData | undefined,
+  factor: number
+): TimelineData | undefined {
+  if (!timeline) {
+    return undefined;
+  }
+
+  // TODO: deep copy
+  const copy = JSON.parse(JSON.stringify(timeline)) as typeof timeline;
+
+  if (copy && copy.data) {
+    let min = copy.data.samples[0];
+    let max = copy.data.samples[0];
+
+    for (let i = 0; i < copy.data.samples.length; i += 1) {
+      const b = copy.data.samples[i];
+
+      if (b < min) {
+        min = b;
+      }
+      if (b > max) {
+        max = b;
+      }
+    }
+
+    const height = 100; // px
+    const skew = (max - min) / height;
+
+    if (copy.data) {
+      copy.data.samples = copy.data.samples.map((a) => {
+        // We don't want to skew negative values, since users are expecting an absent value
+        if (a <= 0) {
+          return 0;
+        }
+
+        // 4 is completely arbitrary, it was eyeballed
+        return a + skew * factor;
+      });
+    }
+  }
+
+  return copy;
+}
+
+function areTimelinesTheSame(
+  timelineA: TimelineData,
+  timelineB?: TimelineData
+) {
+  if (!timelineA || !timelineB) {
+    // for all purposes let's consider two empty timelines the same
+    // since we want to transform them
+    return false;
+  }
+  const dataA = timelineA.data;
+  const dataB = timelineB.data;
+
+  if (!dataA || !dataB) {
+    return false;
+  }
+
+  if (dataA.samples.length !== dataB.samples.length) {
+    return false;
+  }
+
+  const add = (acc: number, a: number) => acc + a;
+
+  // TODO: actually check if they are the same
+  // this is a very poor heuristic
+  const sumA = dataA.samples.reduce(add, 0);
+  const sumB = dataB.samples.reduce(add, 0);
+  return sumA === sumB;
```

Comment:
There are many ways to improve this but I am looking for the simplest one. Any suggestions?